openedx/core/lib/studio_entities.py

```diff
@@ -0,0 +1,64 @@
+"""Studio entity plugin manager and API."""
+import abc
+
+from openedx.core.lib.api.plugins import PluginManager
+
+
+class StudioEntityPluginManager(PluginManager):
+    """Manager for all available Studio entities.
+
+    Examples of Studio entities include Courses, Libraries, and Programs. All Studio
+    entities should implement `StudioEntity`.
+    """
+    NAMESPACE = 'openedx.studio_entity'
+
+    @classmethod
+    def get_enabled_entities(cls):
+        """Returns a list of enabled Studio entities."""
+        entities = cls.get_available_plugins()
+        enabled_entities = [entity for entity in entities.viewvalues() if entity.is_enabled()]
+
+        return enabled_entities
+
+
+class StudioEntity(object):
+    """Abstract class used to represent Studio entities.
+
+    Examples of Studio entities include Courses, Libraries, and Programs.
+    """
+    __metaclass__ = abc.ABCMeta
```

Comment:
Thanks @cpennington. @andy-armstrong I spoke with @cpennington offline to confirm that he agrees with the approach here. He is, since we're not using subclassing of a base class as the plugin registration mechanism.